ts/components/conversation/header/ConversationHeaderSelectionOverlay.tsx

```diff
@@ -36,35 +40,71 @@ export const SelectionOverlay = () => {
   function onCloseOverlay() {
     dispatch(resetSelectedMessageIds());
   }
+  /**
+   * This is a duplicate with the onKeyDown of SessionConversation.
+   * At some point we'll make a global handler to deal with the key presses
+   * and handle them depending on what is visible, but that's not part of this PR
+   */
+  useKey(
+    shouldProcess => {
+      return (
+        shouldProcess.code === 'Escape' ||
+        shouldProcess.code === 'Backspace' ||
+        shouldProcess.code === 'Delete'
+      );
+    },
+    event => {
+      const selectionMode = !!selectedMessageIds.length;
+      switch (event.key) {
+        case 'Escape':
+          if (selectionMode) {
+            onCloseOverlay();
+          }
+          return true;
+        case 'Backspace':
+        case 'Delete':
+          if (selectionMode && selectedConversationKey) {
+            void deleteMessagesForX(selectedMessageIds, selectedConversationKey, isPublic);
+          }
+          return true;
+        default:
+      }
+      return false;
+    }
+  );
 
   const isOnlyServerDeletable = isPublic;
 
+  const classNameAndId = 'message-selection-overlay';
```

Comment:
I'm thinking we should use the convention of `fallbackFocusId`?

Reply:
well we can't have the same id twice in the document right. If we used the same one for the SelectionOverlay and the dialog, when they are both displayed it's going to be messy

Reply:
All good. I misread that classNameAndID is used for the styles and id as well